mipengine/controller/algorithm_executor_nodes.py

```diff
@@ -97,62 +99,15 @@ def __init__(
         request_id: str,
         context_id: str,
         node_tasks_handler: INodeTasksHandler,
-        initial_view_tables_params: Dict[str, Any] = None,
     ):
-        self._node_tasks_handler = node_tasks_handler
-        self.node_id = self._node_tasks_handler.node_id
-        self.request_id = request_id
-        self.context_id = context_id
-
-        self._initial_view_tables = None
-        if initial_view_tables_params is not None:
-            self._initial_view_tables = self._create_initial_view_tables(
-                initial_view_tables_params
-            )
+        self._node_tasks_handler: INodeTasksHandler = node_tasks_handler
+        self.node_id: str = self._node_tasks_handler.node_id
+        self.request_id: str = request_id
+        self.context_id: str = context_id
```

Comment:
They are very useful with type checkers etc. Assigning the wrong type will highlight it as an error and everywhere inside the class you can quickly check what type these variables are.

Reply:
apart from the line
```
        self.node_id: str = self._node_tasks_handler.node_id
```
the other type annotations are not needed for static type checking. The check will catch the error(s) before the assignment to the `self` variables, as soon as the `__init__` is called with wrong  types.

Reply:
I see your point, but isn't it simpler just to add it everywhere instead of having to check if it's automatically generated or not?

If things change in the future we will again have to check if it's automatically generated, which we will most probably forget.

Reply:
it is not my point, it is how static type checking works. 
Even the line
```
self.node_id: str = self._node_tasks_handler.node_id
``` 
that I said it could be left there, should not actually be there, if the type hints in INodeTasksHandler are all set correctly (in all class attributes and methods), `self.node_id` will be correctly inferred from that.
Also what do you mean "if it is automatically generated"? what would be automatically generated? I think you re using pyCharm, there is a mypy plugin that does these checks continually on the fly. 
If things change in the future how?
Adding this everywhere (what does everywhere actually mean?) would create more visual noise in the code, which we already have more than enough, without any benefit. We just need to keep adding type annotations to our functions/methods inputs and outputs for static type checking to work properly, there might be some cases where a variable's type annotation cannot be inferred but these variables are not such a case.

Reply:
We don't have a "team" convention that we must add type annotations to functions/methods or to class initialization parameters. It's not something that everyone does and it's currently not something that we can rely on.

"if it is automatically generated" means that if I change something, for example a method that initializes the variable of a class, maybe the type is inferred, maybe not. I don't know how the type checkers work and if we want to be sure we have to reasearch how they work and IF they work the same way, pyCharm/mypy.

With "everywhere" I meant in all the variables of the class.

If we set and enforce a convention for type annotations in methods etc, after we studied how the type checkers work to cover all cases, then I think we can safely remove them from variables in classes. Until then I think it's safer to be explicit.

Reply:
so you re basically saying that because you do not know how static type checking works, you re just going to add type annotation wherever you see fit, I see. I rest my case.